web-frontend/src/pages/TransferCoins/ReceiveCoins/GenerateLink.tsx

```diff
@@ -0,0 +1,76 @@
+import {
+  Button,
+  Input,
+  MassaToken,
+  PopupModal,
+  PopupModalContent,
+  PopupModalHeader,
+  Selector,
+} from '@massalabs/react-ui-kit';
+import { FiArrowUpRight } from 'react-icons/fi';
+import { formatStandard } from '../../../utils/MassaFormating';
+import CopyContent from './CopyContent';
+
+function GenerateLink({ ...props }) {
+  const {
+    account,
+    modal,
+    url,
+    setModal,
+    setAmount,
+    setProvider,
+    handleGenerate,
```

Comment:
I think we only need account in props.

All the rest can be handled in that component. You can generate the link here, based on account from props for recipient, amount from the form.

```suggestion
```

In addition, why setModal and modal are outside of this component? This make this component heavily coupled with the one containning it

Reply:
I removed setAmount,setProvider, handleGenerate,
    but I still need the url because I am using it to generate the qr code outside of the modal